libs/designsystem/badge/src/badge.component.scss

```
@@ -27,6 +27,42 @@
       left: 0;
     }
   }
+
+  // Colors
+  &.success {
+    ion-badge {
+      --background: #{utils.get-color('success')};
+      --color: #{utils.get-color('success-contrast')};
+    }
+  }
+
+  &.warning {
+    ion-badge {
+      --color: #{utils.get-color('warning-contrast')};
+    }
+
+    &.sm {
+      ion-badge {
+        --background: #{utils.get-color('warning-shade')};
+      }
+    }
+
+    &.md {
+      ion-badge {
+        --background: #{utils.get-color('warning')};
+        @include utils.slotted(kirby-icon) {
+          --background: #{utils.get-color('warning-tint')};
+        }
+      }
+    }
+  }
+
+  &.danger {
+    ion-badge {
+      --background: #{utils.get-color('danger')};
+      --color: #{utils.get-text-color('white')};
+    }
+  }
```

Consider flattening as much as possible now that we are at it:
```suggestion
  // Colors
  &.success ion-badge {
    --background: #{utils.get-color('success')};
    --color: #{utils.get-color('success-contrast')};
  }

  &.warning ion-badge {
    --color: #{utils.get-color('warning-contrast')};
    --background: #{utils.get-color('warning')};
  }

  &.warning.sm ion-badge {
    --background: #{utils.get-color('warning-shade')};
  }
  
  &.warning.md ion-badge {
      @include utils.slotted(kirby-icon) {
          --background: #{utils.get-color('warning-tint')};
        }
  }

  &.danger ion-badge {
    --background: #{utils.get-color('danger')};
    --color: #{utils.get-text-color('white')};
  }

```